samples/core/callback/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2023 The Khronos Group Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+find_package(Threads)
+
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/threads.c"
+"#include <threads.h>
+#include <time.h>
+int main(void) { thrd_sleep(&(struct timespec){.tv_nsec=1}, NULL); }
+")
+
+# Signature can be modernized at CMake version 3.25
+try_compile(
+    HAS_C11_THREADS
+    "${CMAKE_CURRENT_BINARY_DIR}"
+    SOURCES "${CMAKE_CURRENT_BINARY_DIR}/threads.c"
+    C_STANDARD 11
+    C_STANDARD_REQUIRED ON
```

Comment:
Can we simplify this logic if we require C11?  We're already requiring C++14, so perhaps this won't be controversial.

Reply:
Unfortunately no. This very convoluted logic to determine whether the compiler supports C11 threads was required, due to the varying compiler support across the platforms we compile to.

Most notably MSVC doesn't have it, and also it cannot report C standard conformance to CMake properly (see [C_STANDARD_REQUIRED](https://cmake.org/cmake/help/latest/prop_tgt/C_STANDARD_REQUIRED.html)). This is the reason why not having C11 threads only emits a warning, and is not an error.

A proper C11 threads implementation [is coming to MSVC](https://devblogs.microsoft.com/cppblog/c11-threads-in-visual-studio-2022-version-17-8-preview-2/), but at the time of writing this code, it was not available in the CI runners, nor we can assume that everyone has that version of Visual Studio.

Reply:
Oof, what a pain.  This is why we can't have nice things :smile: 

Thanks for the explanation, I'll just resolve this conversation.